src/chronify/time_configs.py

```diff
@@ -152,31 +145,50 @@ def list_timestamps(self) -> list[Any]:
 
         Returns
         -------
-        list[datetime]
+        list[Any]
         """
         return list(self.iter_timestamps())
 
+    @abc.abstractmethod
+    def list_timestamps_from_dataframe(self, df: pd.DataFrame) -> list[Any]:
+        """Return a list of timestamps present in DataFrame.
+        Type of the timestamps depends on the class.
+
+        Returns
+        -------
+        list[Any]
+        """
+
     @abc.abstractmethod
     def iter_timestamps(self) -> Generator[Any, None, None]:
         """Return an iterator over all time indexes in the table.
         Type of the time is dependent on the class.
         """
 
-    def convert_database_timestamps(self, cur: CursorResult) -> list[Any]:
-        """Convert timestamps from the database."""
-        return [x[0] for x in cur]
-
 
 class DatetimeRange(TimeBaseModel):
     """Defines a time range that uses Python datetime instances."""
 
     time_type: Literal[TimeType.DATETIME] = TimeType.DATETIME
-    time_zone: TimeZone
-    start: datetime
-    frequency: timedelta
-    time_based_data_adjustment: TimeBasedDataAdjustment
-    interval_type: TimeIntervalType
-    measurement_type: MeasurementType
+    time_zone: Annotated[
+        Optional[TimeZone],
+        Field(
+            description="Time zone if the timestamps are timezone-aware. "
+            "If None, timestamps are timezone-naive.",
+        ),
+    ] = None
+    start: datetime  # TODO: what if the time zone is specified here?
```

Comment:
While this is supported in other packages like pandas, IMO, it can be ambiguous here because:
If only offset is specified, we can only assume the same offset throughout, making it a standard time assumption.

That said, we can support that as a future feature?

Reply:
This behavior is now different. There is no longer a time_zone field. The user has to specify the time zone, if any, in this field. Please raise any concerns.